src/features/wallet/send/SendTransferForm.tsx

```diff
@@ -0,0 +1,420 @@
+import React, { useEffect, useState } from 'react'
+import { ActivityIndicator } from 'react-native'
+import { useTranslation } from 'react-i18next'
+import { Address } from '@helium/crypto-react-native'
+import { Account } from '@helium/http'
+import Balance, { CurrencyType, NetworkTokens } from '@helium/currency'
+import { useAsync } from 'react-async-hook'
+import animalName from 'angry-purple-tiger'
+import InputField from '../../../components/InputField'
+import TouchableOpacityBox from '../../../components/TouchableOpacityBox'
+import Box from '../../../components/Box'
+import Text from '../../../components/Text'
+import QrCode from '../../../assets/images/qr.svg'
+import Check from '../../../assets/images/check.svg'
+import { useColors } from '../../../theme/themeHooks'
+import LockedField from '../../../components/LockedField'
+import { SendTransfer, SendType } from './sendTypes'
+import { Transfer } from '../../hotspots/transfers/TransferRequests'
+import { decimalSeparator, groupSeparator, locale } from '../../../utils/i18n'
+import { ensLookup } from '../../../utils/explorerClient'
+import useHaptic from '../../../utils/useHaptic'
+import {
+  calculateBurnTxnFee,
+  calculatePaymentTxnFee,
+  calculateTransferTxnFee,
+  useFees,
+} from '../../../utils/fees'
+
+type Props = {
+  account?: Account
+  isLocked: boolean
+  isSeller?: boolean
+  lastReportedActivity?: string
+  onScanPress: () => void
+  sendTransfer: SendTransfer
+  transferData?: Transfer
+  type: SendType
+  updateTransfer: (transferId: number, updates: any) => void
+}
+
+const SendTransferForm = ({
+  account,
+  isLocked,
+  isSeller,
+  lastReportedActivity,
+  onScanPress,
+  sendTransfer,
+  transferData,
+  type,
+  updateTransfer,
+}: Props) => {
+  // Hook init
+  const { t } = useTranslation()
+  const { feeToHNT } = useFees()
+  const { triggerNavHaptic } = useHaptic()
+  const { primaryMain } = useColors()
+
+  // State init
+  const [address, setAddress] = useState<string>(sendTransfer.address)
+  const [addressAlias, setAddressAlias] = useState<string>(
+    sendTransfer.addressAlias,
+  )
+  const [addressLoading, setAddressLoading] = useState(
+    sendTransfer.addressLoading,
+  )
+  const [amount, setAmount] = useState<string>(sendTransfer.amount)
+  const [balanceAmount, setBalanceAmount] = useState<Balance<NetworkTokens>>(
+    sendTransfer.balanceAmount,
+  )
+  const [dcAmount, setDcAmount] = useState<string>(sendTransfer.dcAmount)
+  const [memo, setMemo] = useState<string>(sendTransfer.memo)
+  const [fee, setFee] = useState<Balance<NetworkTokens>>(sendTransfer.fee)
+
+  useEffect(() => {
+    updateTransfer(sendTransfer.id, {
```

Comment:
Looks like `SendTransfer` object has a string id, but the `updateTransfer` functions id parameter is a number.